pkg/util/cert/cert.go

```diff
@@ -99,3 +167,97 @@ func deriveWebhookBaseName(webhookServiceName string) string {
 func buildWebhookConfigurationName(baseName, webhookType string) string {
 	return fmt.Sprintf("%s-%s-webhook-configuration", baseName, webhookType)
 }
+
+func discoverConversionCRDs(ctx context.Context, mgr ctrl.Manager) []string {
+	log := ctrl.Log.WithName("cert")
+	clientset, err := apiextensionsclient.NewForConfig(mgr.GetConfig())
+	if err != nil {
+		log.Error(err, "failed to create clientset")
+		return nil
+	}
+
+	crds, err := clientset.ApiextensionsV1().CustomResourceDefinitions().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		log.Error(err, "failed to list CRDs")
+		return nil
+	}
+
+	var names []string
+	for _, crd := range crds.Items {
+		if crd.Spec.Group == conversionGroup &&
+			crd.Spec.Conversion != nil &&
+			crd.Spec.Conversion.Strategy == apiextensionsv1.WebhookConverter &&
+			crd.Spec.Conversion.Webhook != nil &&
+			crd.Spec.Conversion.Webhook.ClientConfig != nil {
+			names = append(names, crd.Name)
+		}
+	}
+	sort.Strings(names)
+	log.Info("Discovered conversion CRDs", "count", len(names), "names", names)
+	return names
+}
+
+func waitForCAData(mgr ctrl.Manager, cfg config.Configuration, log logr.Logger) ([]byte, error) {
+	if cfg.InternalCertManagement == nil || cfg.InternalCertManagement.WebhookSecretName == nil || cfg.Namespace == nil {
+		return nil, errors.New("cert management disabled")
+	}
+
+	coreClient, err := kubernetes.NewForConfig(mgr.GetConfig())
+	if err != nil {
+		log.Error(err, "failed to create clientset")
+		return nil, err
+	}
+
+	secretKey := types.NamespacedName{
+		Namespace: *cfg.Namespace,
+		Name:      *cfg.InternalCertManagement.WebhookSecretName,
+	}
+
+	for range [30]int{} {
+		secret, err := coreClient.CoreV1().Secrets(secretKey.Namespace).Get(context.Background(), secretKey.Name, metav1.GetOptions{})
+		if err == nil {
+			if data, ok := secret.Data["ca.crt"]; ok && len(data) > 0 {
+				return append([]byte(nil), data...), nil
+			}
+		} else if !apierrors.IsNotFound(err) {
+			log.Error(err, "failed to fetch secret")
+			return nil, err
+		}
+		time.Sleep(2 * time.Second)
+	}
+	err = errors.New("timeout waiting for CA certificate")
+	log.Error(err, "CA bundle injection failed")
+	return nil, err
+}
+
+func injectCABundlesIntoCRDs(mgr ctrl.Manager, crdNames []string, caData []byte, log logr.Logger) error {
```

Comment:
If we are starting the bootstrap manager, then I think we could register the cert rotator into it. And It would do all the retrying for us. Maybe I'm missing something, but it would allow to drop a lot of the  code.